### PR TITLE
Handle captcha

### LINKED ISSUE #11: Captcha response
- [ ] Retry request in case of captcha response

### PR DESCRIPTION
**Pull request by vladimyr**
Making groundwork for #11 🚧 

**UPDATE:** This resolves #11 🎉

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -52,48 +54,82 @@ const selectors = {
...
-    return this.parseAuthorProfile(result.body);
+    const resp = await this.request(url);
+    return this.parseAuthorProfile(resp.body);
   }
 
   async getPubAuthors(query) {
     const { authors } = await this.searchPub(query);
-    if (!authors) {
-      return;
-    }
```
Comment (vladimyr) [inline index.js:87]: `authors` can't be falsy value. If authors aren't found, an empty array gets returned.

**Comment by vladimyr** [comment]
**UPDATE:** This resolves #11 🎉

File: index.js [hunk 2]
```diff
@@ -51,49 +55,106 @@ const selectors = {
...
+  _setupClient() {
+    return got.extend({
+      handlers: [
+        (options, next) => {
+          if (options.isStream) return next(options);
+          return (async () => {
+            const resp = await next(options);
+            const $html = cheerio.load(resp.body);
+            const body = this._verifyPage($html);
+            return Object.assign(resp, { body });
```
Comment (vladimyr) [inline index.js:81]: Swap textual `body` with `Cheerio` instance.

File: index.js [hunk 3]
```diff
@@ -51,49 +55,106 @@ const selectors = {
...
+  init(apiKey, { retries } = {}) {
+    const useProxy = !apiKey;
+    this.useProxy = useProxy;
+
+    let client = this._setupClient();
+    if (useProxy) client = this._setupProxy(client, apiKey);
+    this.client = client;
+
+    if (retries === undefined) retries = useProxy ? 2 : 0;
+    this.retries = Math.min(0, retries);
```
Comment (ikovac) [inline index.js]: I don't get why you are returning min of zero and `retries`? For example, if `retries = 2`, `this.retries` will become 0 ?

File: index.js [hunk 4]
```diff
@@ -51,49 +55,106 @@ const selectors = {
 };
 
 class Scholar {
-  init(key) {
-    if (!key) return this;
-    this.isProxy = true;
-    client.defaults.options = got.mergeOptions(client.defaults.options, {
+  init(apiKey, { retries } = {}) {
+    const useProxy = !apiKey;
```
Comment (ikovac) [inline index.js]:
```suggestion
    const useProxy = !!apiKey;
```
Does `useProxy` need to be `true` if apiKey is set?

Comment (vladimyr) [reply index.js]: Absolutely, this is a rather stupid mistake :facepalm:

File: index.js (hunk 3 above)
Comment (vladimyr) [reply index.js]: Because I'm an idiot. What I've should've done is use `Math.max` to ensure no negative retry count could be set.

Comment (ikovac) [reply index.js]:
```suggestion
    this.retries = Math.max(0, retries);
```

**Review by ikovac — APPROVED** [review]

**Pull request merged**